script.js

```diff
@@ -0,0 +1,168 @@
+  localStorage.setItem(yetText, JSON.stringify(arr));
+};
+// 함수2, setValue in localStorage
+const setDoneLocal = (arr) => {
+  localStorage.setItem(doneText, JSON.stringify(arr));
+};
+
+// 함수3, render in yetList
+const setYet = (obj) => {
+  // DOM 조작(생성)
+  const li = document.createElement('li');
+  const span = document.createElement('span');
+  const delImg = document.createElement('img');
+  const yetBtn = document.createElement('button');
+  // span.innerText = obj.contents;
+  delImg.src = './img/bin.png';
+  yetBtn.className = yetBtnText;
+  yetBtn.innerText = obj.contents;
+  li.id = obj.id;
+  li.append(yetBtn);
+  li.append(span);
+  li.append(delImg);
+  yetList.append(li); // dom 삽입
+  // del , done 처리
+
+  // 숫자 업데이트
+
+  delImg.addEventListener('click', del);
+  yetBtn.addEventListener('click', done);
+
+  updateNum();
+};
+
+// 함수4, render in doneList
+const setDone = (obj) => {
+  const li = document.createElement('li');
+  const span = document.createElement('span');
+  span.innerText = obj.contents;
+  li.id = obj.id;
+  li.append(span);
+  doneList.append(li);
+  updateNum();
+};
+
+// 함수 5, delete data both render and localStorage
+const del = (e) => {
+  // 1. delete rendering data in yet
+  e.target.parentElement.remove();
+
+  // 2. delete localStorage data in yet
+  const id = e.target.parentElement.id;
+  let temp = JSON.parse(localStorage.getItem(yetText));
+  temp = temp.filter((i) => i.id != id);
+  localStorage.setItem(yetText, JSON.stringify(temp));
+  updateNum();
+};
+// 함수 6, delete data both render and localStorage
+const done = (e) => {
+  const id = e.target.parentElement.id;
+  const text = e.target.parentElement.innerText.slice(4, -1);
+  // 1. delete render
+  e.target.parentElement.remove();
+
+  // 2.localStorage
+  // 2.1. delete yet in localStorage
+
+  const temp = JSON.parse(localStorage.getItem(yetText));
+  const yet = temp.filter((i) => i.id != id);
+  const done = temp.filter((i) => i.id == id);
+  localStorage.setItem(yetText, JSON.stringify(yet));
+  // 2.2. add done in localStorage
+
+  const obj = {
+    contents: done[0].contents,
+    id: done[0].id,
+  };
+  doneArr.push(obj);
+  setDoneLocal(doneArr);
+  // 3. render (함수4)
+  setDone2();
+
+  // 4. update num
+  updateNum();
+};
+
+const getTodoNum = () => {
+  return JSON.parse(localStorage.getItem(yetText))?.length;
+};
+
+const getDoneNum = () => {
+  return JSON.parse(localStorage.getItem(doneText))?.length;
+};
+
+const setDone2 = () => {
+  const li = document.createElement('li');
+  const span = document.createElement('span');
+  const temp = JSON.parse(localStorage.getItem(doneText));
+  temp.map((i) => {
+    span.innerText = i.contents;
+    li.id = i.id;
+    li.append(span);
+  });
+  doneList.append(li);
+};
+
+const updateNum = () => {
+  const a = getTodoNum();
+  const b = getDoneNum();
+  todoNum.innerText = `To Do (${a == undefined ? 0 : a})`;
+  doneNum.innerText = `DONE (${b == undefined ? 0 : b})`;
```

Comment:
이 부분도 `a`, `b` 말고 직관적인 변수명으로 바꿔 주세요! 또한 아래처럼 작성하셔도 똑같이 동작합니다.

```suggestion
  const a = getTodoNum();
  const b = getDoneNum();
  todoNum.innerText = `To Do (${a})`;
  doneNum.innerText = `DONE (${b})`;
```